core/src/ldml/ldml_processor.cpp

```diff
@@ -195,9 +195,10 @@ ldml_processor::process_event(
         return KM_KBP_STATUS_OK; // Nothing to do- no key
       }
       const std::u16string &str = key->second;
-      for(size_t i=0; i<str.length(); i++) {
-        state->context().push_character(str[i]);
-        state->actions().push_character(str[i]);
+      const std::u32string str32 = kmx::u16string_to_u32string(str);
+      for(size_t i=0; i<str32.length(); i++) {
+        state->context().push_character(str32[i]);
+        state->actions().push_character(str32[i]);
```

Comment:
Surprisingly, this fix is actually _not needed_. It seems to have no effect on the output text.

Reply:
in any event, …
aha! deletion event.  Backspace breaks it… testing

Reply:
yup that was it.